### ISSUE TITLE
Submit storm topology in local 

### ISSUE THREAD

**Issue by isspek** [DigitalPebble/storm-crawler#441]
I created a package  containing my custom storm topology for my web-app project. It runs in local mode without any problem.  However when  I call this method `ConfigurableTopology.start(new CrawlTopology(), args);` as local mode in the package, from the rest controller based on spring framework, I receive the following errors: 

```
152504 [Thread-9] ERROR o.a.s.event - Error when processing event
java.io.FileNotFoundException: Source '...\wtpwebapps\project-name\WEB-INF\lib\cdm-4.5.5.jar!\resources' does not exist
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.checkFileRequirements(FileUtils.java:1405) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1368) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1268) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1237) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.daemon.supervisor$fn__9436.invoke(supervisor.clj:1178) ~[storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.MultiFn.invoke(MultiFn.java:243) ~[clojure-1.7.0.jar:?]
    at org.apache.storm.daemon.supervisor$mk_synchronize_supervisor$this__9163$fn__9181.invoke(supervisor.clj:579) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.daemon.supervisor$mk_synchronize_supervisor$this__9163.invoke(supervisor.clj:578) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.event$event_manager$fn__8735.invoke(event.clj:40) [storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.AFn.run(AFn.java:22) [clojure-1.7.0.jar:?]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
152504 [Thread-11] ERROR o.a.s.event - Error when processing event
java.io.FileNotFoundException: Source '...\project-name\WEB-INF\lib\cdm-4.5.5.jar!\resources' does not exist
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.checkFileRequirements(FileUtils.java:1405) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1368) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1268) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.shade.org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1237) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.daemon.supervisor$fn__9436.invoke(supervisor.clj:1178) ~[storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.MultiFn.invoke(MultiFn.java:243) ~[clojure-1.7.0.jar:?]
    at org.apache.storm.daemon.supervisor$mk_synchronize_supervisor$this__9163$fn__9181.invoke(supervisor.clj:579) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.daemon.supervisor$mk_synchronize_supervisor$this__9163.invoke(supervisor.clj:578) ~[storm-core-1.0.2.jar:1.0.2]
    at org.apache.storm.event$event_manager$fn__8735.invoke(event.clj:40) [storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.AFn.run(AFn.java:22) [clojure-1.7.0.jar:?]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
152508 [Thread-11] ERROR o.a.s.util - Halting process: ("Error when processing an event")
java.lang.RuntimeException: ("Error when processing an event")
    at org.apache.storm.util$exit_process_BANG_.doInvoke(util.clj:341) [storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.7.0.jar:?]
    at org.apache.storm.event$event_manager$fn__8735.invoke(event.clj:48) [storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.AFn.run(AFn.java:22) [clojure-1.7.0.jar:?]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
152508 [Thread-9] ERROR o.a.s.util - Halting process: ("Error when processing an event")
java.lang.RuntimeException: ("Error when processing an event")
    at org.apache.storm.util$exit_process_BANG_.doInvoke(util.clj:341) [storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.7.0.jar:?]
    at org.apache.storm.event$event_manager$fn__8735.invoke(event.clj:48) [storm-core-1.0.2.jar:1.0.2]
    at clojure.lang.AFn.run(AFn.java:22) [clojure-1.7.0.jar:?]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
2017-03-20 15:47:21,335 FATAL Unable to register shutdown hook because JVM is shutting down.
154735 [Thread-10] INFO  o.a.s.d.supervisor - Removing code for storm id crawl-1-1490010427
```

I just want to use storm crawler in fat jar package, not to install it in neither my local system nor server. I am not sure this issue is related to storm or storm crawler. I apreciate the any ideas for resolving this issue.

**Comment by jnioche** [comment]
See #324, this is a Storm-related issue. I opened a JIRA https://issues.apache.org/jira/browse/STORM-2055, it could be related to https://issues.apache.org/jira/browse/STORM-2324, which in theory has been fixed by Storm 1.0.3
I opened #437 for upgrading to 1.0.3 but got into problems with it.  Since you are running in local mode, you probably won't have the same problem. Could you try changing the versions of Storm to 1.0.3 and see if it fixes it?

**Comment by jnioche** [comment]
Investigated a bit, you can run it in local mode without getting the error above by upgrading the storm dependency to 1.0.3.

**Comment by jnioche** [comment]
@isspek I have upgraded the version of Storm to 1.0.3 in the master branch. Could you please recompile it and use it in your project? Please let me know if it fixes the issue. Thanks

**Comment by isspek** [comment]
@jnioche Sorry I couldn't reply because I am away from that computer. When I upgraded to Storm 1.0.3, I didn't this error, but I received new exception related to windows. I searched the internet and the problem seems related to the version of Windows (that computer is XP), and Eclipse.  To be sure,  I am trying to simulate the problem with machine installed Windows 10.  I will update this comment tomorrow with new exception message and the solution if I come up.
